src/pybind/mgr/ansible/module.py

```diff
@@ -455,6 +566,45 @@ def process_playbook_result(inventory_events, ar_client, playbook_uuid):
 
     return result
 
+def process_hosts_ls(host_ls_json):
+    """ Format the output of host ls call
+
+    Ansible runner service provides the following json:
+            {
+            "status": "OK",
+            "msg": "",
+            "data": {
+                "hosts": [
+                    "host_a",
+                    "host_b",
+                    ...
+                    "host_x",
+                    ]
+                }
+            }
+
+    :return: list of InventoryNodes
+    """
+
+    inventory_nodes = []
+
+    try:
+        json_resp = json.loads(host_ls_json)
+
+        for host in json_resp["data"]["hosts"]:
+            inventory_nodes.append(orchestrator.InventoryNode(host, []))
+
+    except ValueError:
+        self.log.exception("Malformed json response")
```

Comment:
would it make sense to re-raise the exception here? We're logging important debug into the log file. Would it make sense to actually pass this info to the use?

Reply:
there is no `self` here.

Reply:
This function is assigned  to "process_host_ls" attribute in completion clasess, and this classes have a log attribute.

Reply:
Fixed. Now the "process_ls" method provided as output any kind of problem parsing json.

Reply:
> there is no `self` here.

still not fixed.